numba_dpex/experimental/_kernel_dpcpp_spirv_overloads/_index_space_id_overloads.py

```diff
@@ -58,10 +83,16 @@ def ol_item_get_id(item, dim):
         TypingError: When argument is not an integer.
     """
     if not isinstance(item, ItemType):
-        raise TypingError("Only Item is supported")
+        raise TypingError(
+            "Expected an item should to be an Item value, but "
+            f"encountered {type(item)}"
+        )
 
     if not isinstance(dim, types.Integer):
-        raise TypingError("Only integers supported")
+        raise TypingError(
```

Comment:
Can `dim` be a variable? In other words, would the below code work?
```
dim = 0
id = item.get_id(dim)
```

Reply:
The snippet should work as numba will infer the type of `dim` as types.Integer